app/renderer/components/main/siteInfo.js

```diff
@@ -3,259 +3,271 @@
-          <span className={cx({
-            fa: true,
-            'fa-unlock': true,
-            [css(styles.secureIcon__fa)]: true
-          })} />
-          <span className={css(styles.secureIcon__label)}
-            data-test-id='partiallySecureConnection'
-            data-l10n-id='partiallySecureConnection'
-          />
-        </div>
+      return <div className={css(styles.secureIcon__wrapper)}>
+        <span className={cx({
+          fa: true,
+          'fa-unlock': true,
+          [css(styles.secureIcon__fa)]: true
+        })} />
+        <span
+          className={css(styles.secureIcon__label)}
+          data-test-id='partiallySecureConnection'
+          data-l10n-id='partiallySecureConnection'
+        />
+      </div>
     } else {
       // insecure
-      secureIcon =
-        <div className={css(styles.secureIcon__wrapper)}>
-          <span className={cx({
-            fa: true,
-            'fa-unlock': true,
-            [css(styles.secureIcon__fa)]: true
-          })} />
-          <span className={css(styles.secureIcon__label)}
-            data-test-id='insecureConnection'
-            data-l10n-id='insecureConnection'
-          />
-        </div>
+      return <div className={css(styles.secureIcon__wrapper)}>
+        <span className={cx({
+          fa: true,
+          'fa-unlock': true,
+          [css(styles.secureIcon__fa)]: true
+        })} />
+        <span
+          className={css(styles.secureIcon__label)}
+          data-test-id='insecureConnection'
+          data-l10n-id='insecureConnection'
+        />
+      </div>
     }
+  }
 
-    let partitionInfo
-    if (this.partitionNumber) {
-      partitionInfo =
-        <div className={css(styles.secureIcon__wrapper)}>
-          <span className={cx({
-            fa: true,
-            'fa-user': true,
-            [css(styles.secureIcon__fa)]: true
-          })} />
-          <span className={css(styles.secureIcon__label)}
-            data-test-id='partitionNumber'
-            data-l10n-args={JSON.stringify(l10nArgs)}
-            data-l10n-id='sessionInfo'
-          />
-        </div>
+  get partitionInfo () {
+    if (this.props.partitionNumber) {
+      // Figure out the partition info display
+      let l10nArgs = {
+        partitionNumber: this.props.partitionNumber
+      }
+
+      return <div className={css(styles.secureIcon__wrapper)}>
+        <span className={cx({
+          fa: true,
+          'fa-user': true,
+          [css(styles.secureIcon__fa)]: true
+        })} />
+        <span
+          className={css(styles.secureIcon__label)}
+          data-test-id='partitionNumber'
+          data-l10n-args={JSON.stringify(l10nArgs)}
+          data-l10n-id='sessionInfo'
+        />
+      </div>
     }
 
-    let connectionInfo = null
-    let viewCertificateButton = null
+    return null
+  }
 
+  get viewCertificateButton () {
+    // TODO(Anthony): Hide it until muon support linux
```

Comment:
This should have a unit test, IMO. Often times unit tests can document the code as good or better than comments 😄  (the test acting as a specification)